video/out/wayland_common.c

```diff
@@ -1048,10 +1540,36 @@ static void registry_handle_add(void *data, struct wl_registry *reg, uint32_t id
     if (!strcmp(interface, wl_compositor_interface.name) && (ver >= 4) && found++) {
         wl->compositor = wl_registry_bind(reg, id, &wl_compositor_interface, 4);
         wl->surface = wl_compositor_create_surface(wl->compositor);
+        wl->video_surface = wl_compositor_create_surface(wl->compositor);
+        /* never accept input events on the video surface */
+        struct wl_region *region = wl_compositor_create_region (wl->compositor);
+        wl_surface_set_input_region (wl->video_surface, region);
+        wl_region_destroy (region);
         wl->cursor_surface = wl_compositor_create_surface(wl->compositor);
         wl_surface_add_listener(wl->surface, &surface_listener, wl);
     }
 
+    if (!strcmp(interface, wl_subcompositor_interface.name) && (ver >= 1) && found++) {
+         wl->subcompositor = wl_registry_bind(reg, id, &wl_subcompositor_interface, 1);
+         wl->video_subsurface = wl_subcompositor_get_subsurface(wl->subcompositor, wl->video_surface, wl->surface);
+         wl_subsurface_set_desync (wl->video_subsurface);
+    }
+
+	if (!strcmp (interface, zwp_linux_dmabuf_v1_interface.name) && (ver >= 2) && found++) {
+	  wl->dmabuf =
+		  wl_registry_bind (reg, id, &zwp_linux_dmabuf_v1_interface, 2);
```

Comment:
This can be on one line (wayland names are long and we have much worse already).